clarity/src/vm/functions/database.rs

```diff
@@ -203,6 +203,23 @@ pub fn special_contract_call(
         }
     }
 
+    let sender = nested_env.sender.clone();
+    let caller = nested_env.caller.clone();
+
+    let mut function_args = vec![];
+    for arg in args[2..].iter() {
+        let evaluated_arg = eval(arg, env, context)?;
+        function_args.push(evaluated_arg);
+    }
+
+    env.register_contract_call_event(
```

Comment:
My initial plan was to register this event before `contract-call` is executed to maintain correct order of events. However I'm not sure if I could do that without any negative consequences, therefore it is added after.

@jcnelson @lgalabru @kantai I would love to hear your feedback about that.